### ISSUE TITLE
reportLocation does not report location anymore

### ISSUE THREAD

**Issue by metabubble** [owntracks/android#627]
I use owntracks to keep track of my android battery in home assistant when the phone is home, so I have an automation that every 5 minutes calls

```
_type: cmd
action: reportLocation
```

This used to work reliably, but since the update, that changed to the 4 different monitoring modes, when the phone is in "significant changes monitoring mode", it will not respond to reportLocation anymore. Not even when the app is in foreground.

Here is a portion of my mqtt server. The location report was triggered manually from the phone. The request wasn't changed, only thing that changed was the owntracks update.
```
owntracks/christian-pixel3xl/pixel3xl {"_type":"location","acc":15,"alt":0.0,"batt":38,"conn":"w","lat":xxxx,"lon":xxxx,"t":"u","tid":"CH","tst":1544329204,"vac":0,"vel":0.0}
owntracks/christian-pixel3xl/pixel3xl/cmd {"_type":"cmd","action":"reportLocation"}
owntracks/christian-pixel3xl/pixel3xl/cmd {"_type":"cmd","action":"reportLocation"}
owntracks/christian-pixel3xl/pixel3xl/cmd {"_type":"cmd","action":"reportLocation"}

```

**Comment by binarybucks** [comment]
thank you for reporting this. The code was accidently checking for the wrong mode. 
A fixed version will be available soon. 

Out of curiosity, isnt't the 5m interval too much for checking the battery level of the device? The new version can also report every `ping` interval minutes (>= 15 minutes). Maybe you can also use that instead of regularly pinging the device.

**Comment by metabubble** [comment]
I want to stop charging when the battery reaches 75% (for battery longevity). On a fast charger, it already charges 1% per minute, so querying every 5 minutes already gives me a blur of 5%.

At home, it is usually resting on a qi charger during the day, so I can always leave with a battery between 70 and 80%... just don't want to constantly trickle charge at 100% because I want to keep my phone for longer than a year.

**Comment by Natanji** [comment]
Because I am having similar problems: is "significant changes mode" the same as how Owntracks on Android worked before, and the other modes were just added? Or is it different from how that worked? Because I want the same behavior as before the update, it was perfect, I don't need the new modes... but I don't understand yet which mode I need to use.

**Comment by binarybucks** [comment]
You're correct. Significant changes mode is the exactly like the old behaviour and can be tweaked with the same settings. Just the foreground/background things were removed.

**Comment by binarybucks** [comment]
It is now working as intended. Before the update, exactly that should have happened, but for various reasons, new location requests were created once in a while. This lead to new locations being send, even if the device didn't move x meters. 

If you want the "old/broken" behavior, you can set locator displacement to 0 or a small number in the settings. With displacement set to 0, you will get updates roughly every `interval even if the device only moved a bit.

**Comment by Natanji** [comment]
That doesn't seem to restore the old behavior though? Preciously I could set it it to e.g. 500m and 10 minutes and the condition seemed to be an "or" between those. When not moving much, that would make it trigger every 10 minutes. But when e.g. in a car and having a navigation app active, it would trigger more often (every 500m) and give more fine-grained updates. I don't see how this really intelligent and good behavior can be restored in the old version. Can you maybe create a new mode that works like the old version? It has merit. It worked perfectly for me.

**Comment by binarybucks** [comment]
I cannot create a new mode that works like the old version because the actual way how the location is requested hasn't been changed in signifficant mode. A location request has two parameters that influence when it is triggered. An interval and a displacement. With a displacement set to >0, it takes precedence over the interval however, hence my recomendation to disable it. The way those are set is exactly the same as before. There are now just three other modes around it with some different attributes. The only thing that has changed is, that the request is not recreated as often as before. 

With 10 minute interval and 500m displacement a new location should be provided at least if the device moved 500m in 10 minutes. If a new location is available earlier (e.g. other apps such as Google Maps requested a location), updates are provided faster.

**Comment by binarybucks** [comment]
@metabubble the fix is now available on Google Play in 2.1.1

**Comment by vontrapp** [comment]
Then it is NOT the same as it worked before! As before, and I quote, "The only thing that has changed is, that the request is not recreated as often as before."

So before, the request was created more often. Was that more often as in every time the interval time expired, whereas now the interval is a min update interval instead of a max update interval? Or IOW like Natanji says, before the interval and distance parameters were an OR condition. If the distance changed significantly OR the time interval elapsed, then a new update was published.

And now, you're saying that the conditions are AND, if the distance is greater than the configured parameter AND the time elapsed since last update is more than configured parameter, only then an update is sent.

Furthermore, I'm concerned as you say things like "moved 500m in 10 minutes". Does this mean that if I move 1000m in 25 minutes (a very leisurely walk), that I won't get an update at all? Since I never moved 500m within any 10m time span?
